Mixpanel/UIView+MPHelpers.m

```diff
@@ -0,0 +1,213 @@
+    return [UIImage imageWithData:imageData];
+}
+
+// mp_targetActions
+- (NSArray *)mp_targetActions
+{
+    NSMutableArray *targetActions = [NSMutableArray array];
+    if ([self isKindOfClass:[UIControl class]]) {
+        for (id target in [(UIControl *)(self) allTargets]) {
+            UIControlEvents allEvents = UIControlEventAllTouchEvents | UIControlEventAllEditingEvents;
+            for (NSUInteger e = 0; (allEvents >> e) > 0; e++) {
+                UIControlEvents event = allEvents & (0x01 << e);
+                if(event) {
+                    NSArray *actions = [(UIControl *)(self) actionsForTarget:target forControlEvent:event];
+                    NSArray *ignoreActions = @[@"preVerify:forEvent:", @"execute:forEvent:"];
+                    for (NSString *action in actions) {
+                        if ([ignoreActions indexOfObject:action] == NSNotFound)
+                        {
+                            [targetActions addObject:[NSString stringWithFormat:@"%lu/%@", event, action]];
+                        }
+                    }
+                }
+            }
+        }
+    }
+    return [targetActions copy];
+}
+
+// Set by a userDefinedRuntimeAttr in the MPTagNibs.rb script
+- (void)setMixpanelViewId:(id)object
+{
+    objc_setAssociatedObject(self, @selector(mixpanelViewId), [object copy], OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+}
+
+
+// mp_viewId
+//
+- (NSString *)mp_varA
+{
+    return mp_encryptHelper(objc_getAssociatedObject(self, @selector(mixpanelViewId)));
+}
+
+// mp_controllerVariable
+- (NSString *)mp_varB
+{
+    NSString *result = nil;
+    if ([self isKindOfClass:[UIControl class]]) {
+        UIResponder *responder = [self nextResponder];
+        while (responder && ![responder isKindOfClass:[UIViewController class]]) {
+            responder = [responder nextResponder];
+        }
+        if (responder) {
+            uint count;
+            Ivar *ivars = class_copyIvarList([responder class], &count);
+            for (uint i = 0; i < count; i++) {
+                Ivar ivar = ivars[i];
+                if (ivar_getTypeEncoding(ivar)[0] == '@' && object_getIvar(responder, ivar) == self) {
+                    result = [NSString stringWithCString:ivar_getName(ivar) encoding:NSUTF8StringEncoding];
+                    break;
+                }
+            }
+            free(ivars);
+        }
+    }
+    return mp_encryptHelper(result);
+}
+
+/*
+ mp_imageFingerprint
+ 
+ Creates a short string which is a fingerprint of a UIButton's image property.
+ It does this by downsampling the image to 8x8 and then downsampling the resulting
+ 32bit pixel data to 8 bit. This should allow us to select images that are identical or
+ almost identical in appearance without having to compare the whole image.
+ 
+ Returns a base64 encoded string representing an 8x8 bitmap of 8 bit rgba data
+ (2 bits per component).
+ */
+- (NSString *)mp_varC
```

Comment:
can you give all these guys back their original names and just have a section at the bottom where you alias them eg. 

```
- (NSString *)mp_varC { return [self mp_imageFingerprint]; }
- (NSString *)mp_varD { ...
```

We're kinda stuck with these obfuscated names forever because we use them in A/B testing path selectors, but I'd prefer that the code at least become somewhat readable again.